test/ApplyArgs.purs

```diff
@@ -0,0 +1,170 @@
+module Test.Ctl.ApplyArgs (suite, main, contract) where
+
+import Contract.Prelude
+
+import Aeson (Aeson, encodeAeson)
+import Contract.Config (ServerConfig, testnetConfig)
+import Contract.Monad (Contract, launchAff_, runContract, throwContractError)
+import Contract.PlutusData (PlutusData(List, Map, Bytes), toData)
+import Contract.Prim.ByteArray (byteArrayToHex, hexToByteArrayUnsafe)
+import Contract.Scripts (PlutusScript(PlutusScript), Validator(Validator))
+import Contract.Transaction (Language, plutusV1Script)
+import Ctl.Examples.AlwaysSucceeds (alwaysSucceedsScript)
+import Ctl.Examples.IncludeDatum (only42Script)
+import Ctl.Examples.PlutusV2.Scripts.AlwaysSucceeds (alwaysSucceedsScriptV2)
+import Ctl.Internal.ApplyArgs (applyArgs)
+import Ctl.Internal.Helpers ((<</>>))
+import Ctl.Internal.Plutip.Server (stopChildProcessWithPort)
+import Ctl.Internal.Plutip.Spawn
+  ( ManagedProcess
+  , NewOutputAction(NoOp, Success)
+  , spawn
+  )
+import Ctl.Internal.QueryM
+  ( ClientError(ClientEncodingError)
+  , handleAffjaxResponse
+  , mkHttpUrl
+  , postAeson
+  , scriptToAeson
+  )
+import Ctl.Internal.Serialization (toBytes) as Serialization
+import Ctl.Internal.Serialization.PlutusData (convertPlutusData) as Serialization
+import Ctl.Internal.Test.TestPlanM (TestPlanM, interpret)
+import Data.BigInt (fromInt)
+import Data.List.Lazy (replicate)
+import Data.String (Pattern(Pattern))
+import Data.String as String
+import Data.UInt (UInt)
+import Data.UInt as UInt
+import Effect.Aff (bracket)
+import Foreign.Object as Object
+import Mote (group, test)
+import Node.ChildProcess (defaultSpawnOptions)
+import Untagged.Union (asOneOf)
+
+contract ∷ Contract () Unit
+contract = liftEffect main
+
+contractApply :: Array PlutusData -> Contract () Validator -> Contract () Unit
+contractApply params loadScript = do
+  Validator script <- loadScript
+  let escr1 = applyArgs script params
+  escr2 <- applyArgsOld script params
+  applied1 <- either throwContractError pure escr1
+  applied2 <- either throwContractError pure escr2
+  unless (applied1 == applied2) $
+    throwContractError "Applying script result wrong"
+
+contractInvalidApply :: PlutusScript -> Array PlutusData -> Contract () Unit
+contractInvalidApply script param = do
+  case applyArgs script param of
+    Left _ -> pure unit
+    _ -> throwContractError "Accident"
+
+suite :: TestPlanM (Aff Unit) Unit
+suite =
+  group "Applying parameters to script" do
+    traverse_
+      ( test "Result of applying parameters the same for wasm and haskell"
+          <<< runContract testnetConfig
+      )
+      (map (uncurry contractApply) $ ((\x y -> x /\ y) <$> paramss <*> scripts))
+    test "Applying to invalid script returns Left" $
+      runContract testnetConfig (contractInvalidApply invalidScript [ n 1 ])
+
+  where
+  invalidScript = plutusV1Script $ hexToByteArrayUnsafe "ffffff"
+  scripts = [ alwaysSucceedsScript, alwaysSucceedsScriptV2, only42Script ]
```

Comment:
Does it make sense to apply arguments to fully applied script fixtures?

Reply:
doesn't matter, it creates a term like `(Apply (Apply script arg1) arg2)` without evaluation. (haskell does same)